setup.py

```diff
@@ -1,8 +1,22 @@
 """Setup script for lasio"""
 
+import lasio
+import re
 from setuptools import setup
 import os
 
+__init__file = os.path.join('lasio', '__init__.py')
+with open(__init__file, 'r') as f:
+    file_contents = f.read()
+
+changed_file_contents = re.sub(r"(__version__\s?=\s?)[\"\'\w()]+", fr"""\g<1>'{lasio.version()}'""", file_contents)
```

Comment:
This regex is very fancy but quite unnecessary. You could just do one that does: "find '__version__', followed by anything, followed by a newline, and replace the whole thing with '__version__ = blah\n'".

No actual need to use groups, or care about what is on the rest of the line after __version__.

Reply:
I would want to be a bit more specific and include the double _ or it will match a lot of "version" strings spread around the entire file, but I see what you're saying I always tend to unnecessarily overcomplicate every time I use regex Lol